receiver.go

```diff
@@ -401,6 +401,15 @@ func newReceiver(source string, session *Session, opts *ReceiverOptions) (*Recei
 		r.l.linkCredit = 0
 		r.autoSendFlow = false
 	}
+
+	if opts.DesiredCapabilities != nil {
```

Comment:
This will set `r.l.desiredCapabilities` if `opts.DesiredCapabilities` is empty. Is that ok or should this be a `len()` check?

Reply:
It's okay - PerformAttach's marshal method will optimize this before it gets sent so we don't have to do anything special.